pgsql-flexible-server.tf

```diff
@@ -1,25 +1,52 @@
+locals {
+  default_name = var.component != "" ? "${var.product}-${var.component}" : var.product
+  name         = var.name != "" ? var.name : local.default_name
+  server_name  = "${local.name}-${var.env}"
+
+  private_dns_zone_id = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg/providers/Microsoft.Network/privateDnsZones/private.postgres.database.azure.com"
+}
+
+resource "random_password" "password" {
+  length = 20
+  # safer set of special characters for pasting in the shell
+  override_special = "()-_"
+}
+
 resource "azurerm_postgresql_flexible_server" "pgsql_server" {
-  name                = var.pgsql_server_name
-  resource_group_name = var.resource_group_name
+  name                = local.server_name
+  resource_group_name = azurerm_resource_group.rg.name
   location            = var.location
   version             = var.pgsql_version
 
   delegated_subnet_id = var.pgsql_delegated_subnet_id
-  private_dns_zone_id = var.pgsql_private_dns_zone_id
+  private_dns_zone_id = local.private_dns_zone_id
 
   administrator_login    = var.pgsql_admin_username
-  administrator_password = var.pgsql_admin_password
-  zone                   = var.pgsql_server_zone
+  administrator_password = random_password.password.result
 
   storage_mb = var.pgsql_storage_mb
 
   sku_name = var.pgsql_sku
 
-  common_tags = var.common_tags
+  tags = var.common_tags
+
+  high_availability {
+    mode = "ZoneRedundant"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      zone,
+      high_availability.0.standby_availability_zone,
+    ]
+  }
```

Comment:
terraform tries to set these to `null` otherwise, it's recommended to do this in the docs anyway